### PR TITLE
triple support upper header key

### PR DESCRIPTION
**Pull request by EarthChen**
## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Review by AlbumenJ — APPROVED** [review]

**Comment by EarthChen** [comment]
This configuration does not need to be configured separately for each scope value, so only one is placed in the protocol.

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/StreamUtils.java [hunk 1]
```diff
@@ -89,9 +95,23 @@ public static void convertAttachment(DefaultHttp2Headers headers,
             if (TripleHeaderEnum.containsExcludeAttachments(key)) {
                 continue;
             }
+            if (!entry.getKey().equals(key)) {
```
Comment (guohao) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/StreamUtils.java:98]:
```java
if(needConvertHeaderKey)
```

**Review by guohao — APPROVED** [review]
LGTM

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/TripleProtocol.java [hunk 2]
```diff
@@ -57,13 +60,22 @@ public class TripleProtocol extends AbstractProtocol {
...
+
     private boolean versionChecked = false;
 
+
     public TripleProtocol(FrameworkModel frameworkModel) {
         this.frameworkModel = frameworkModel;
         this.triBuiltinService = new TriBuiltinService(frameworkModel);
         this.pathResolver = frameworkModel.getExtensionLoader(PathResolver.class)
             .getDefaultExtension();
+        CONVERT_NO_LOWER_HEADER = ConfigurationUtils.getEnvConfiguration(ApplicationModel.defaultModel())
```
Comment (AlbumenJ) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/TripleProtocol.java:77]: should not use `ApplicationModel.defaultModel()`

Default model is only used when compatible mode.

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/StreamUtils.java [hunk 3]
```diff
@@ -92,6 +98,22 @@ public static void convertAttachment(DefaultHttp2Headers headers,
...
+        if (needConvertHeaderKey) {
+            Map<String, String> needConvertKey = attachments.entrySet()
+                .stream()
+                .filter(it -> !headers.contains(it.getKey()))
+                .collect(Collectors.toMap(Map.Entry::getKey, it -> it.getKey().toLowerCase(Locale.ROOT)));
+            if (!needConvertKey.isEmpty()) {
+                String needConvertJson = JsonUtils.getJson().toJson(needConvertKey);
+                headers.add(TripleHeaderEnum.TRI_HEADER_CONVERT.getHeader(), TriRpcStatus.encodeMessage(needConvertJson));
+            }
+        }
```
Comment (AlbumenJ) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/stream/StreamUtils.java:110]: log out to notify users

**Pull request merged**